app/src/main/java/com/suntraining/aesport/ui/binding/RecyclerViewBinding.kt

```diff
@@ -0,0 +1,14 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package com.suntraining.aesport.ui.binding
+
+import androidx.databinding.BindingAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.suntraining.aesport.base.UpdateData
+
+@BindingAdapter("app:data")
+fun <T> setDataRecyclerView(recyclerView: RecyclerView, data: List<T>?) {
+    if (recyclerView.adapter is UpdateData<*>) {
+        (recyclerView.adapter as UpdateData<T>).updateData(data)
+    }
+}
```

Comment:
scope của Project 2 bé, ít hàm data binding nên cũng k cần phải tách ra nhiều file, để 1 file sẽ dễ quản lý hơn 
```suggestion
@BindingAdapter("app:data")
fun <T> RecyclerView.setDataRecyclerView(data: List<T>?) {
    if (recyclerView.adapter is UpdateData<*>) {
        (recyclerView.adapter as UpdateData<T>).updateData(data)
    }
}
```